Sources/MapboxMaps/Foundation/MapView+Lifecycle.swift

```diff
@@ -0,0 +1,89 @@
+import Foundation
+import UIKit
+
+@available(iOSApplicationExtension, unavailable)
+extension MapView {
+    private static let UIApplicationSceneManifestKey = "UIApplicationSceneManifest"
+
+    internal func subscribeToMemoryWarningNotification() {
+        NotificationCenter.default.addObserver(self,
+                                               selector: #selector(didReceiveMemoryWarning),
+                                               name: UIApplication.didReceiveMemoryWarningNotification,
+                                               object: nil)
+    }
+
+    internal func subscribeToLifecycleNotifications() {
+        if Bundle.main.object(forInfoDictionaryKey: MapView.UIApplicationSceneManifestKey) != nil {
+            if #available(iOS 13.0, *) {
+                subscribeToSceneLifecycleNotifications()
+            }
+        } else {
+            subscribeToApplicationsLifecycleNotifications()
+        }
+    }
+
+    internal func unsubscribeFromLifecycleNotifications() {
+        if Bundle.main.object(forInfoDictionaryKey: MapView.UIApplicationSceneManifestKey) != nil {
+            if #available(iOS 13.0, *) {
+                NotificationCenter.default.removeObserver(self, name: UIScene.willEnterForegroundNotification, object: nil)
+                NotificationCenter.default.removeObserver(self, name: UIScene.didEnterBackgroundNotification, object: nil)
+            }
+        } else {
+            NotificationCenter.default.removeObserver(self, name: UIApplication.willEnterForegroundNotification, object: nil)
+            NotificationCenter.default.removeObserver(self, name: UIApplication.didEnterBackgroundNotification, object: nil)
+        }
```

Comment:
This can be simplified. No need to check for UIApplicationSceneManifestKey. Just unsubscribe from all 4 (or 2 if < iOS 13) every time.

Reply:
Good point, indeed.